revolv/payments/utils.py

```diff
@@ -1,6 +1,398 @@
+        """
+        Creates and fills in the 'repayments', 'reinvestments', and 'donations' subdictionary of 'cash_in' and 'cash_out'
+        within the dictionary for a certain time period and project.
+
+        :args:
+            filtered_dict: The dictionary containing all the information.
+            date: A string represenation of temp_start_date.
+            proj: The project of interest.
+            temp_start_date: The start date of this time period.
+            temp_end_date: The end date of this time period.
+            revolv_earnings_rate: A float from 0-1 representing what percentage of repayments RE-volv takes as earnings.
+            payment_service_fee_rate: A float from 0-1 representing what percentage of donations that are taken by payment service fees.
+        """
+        self.set_repayment_table_values(filtered_dict, date, proj, temp_start_date, temp_end_date, revolv_earnings_rate)
+        self.set_donation_table_values(filtered_dict, date, proj, temp_start_date, temp_end_date, payment_service_fee_rate)
+        self.set_reinvestment_table_values(filtered_dict, date, proj, temp_start_date, temp_end_date)
+
+    def aggregate(self):
+        """
+        Creates and returns a dictionary with the structure shown below.
+        It filters data using the instance variables self.start_date, self.end_date, 
+        and self.project_filter.    
+    
+        The data is structured as shown below:
+        data = {
+            'cash_in': {
+                'adjustments': {
+                    'Surprise Income': [0, 0, 5],
+                },
+                'donations': {
+                    'Power Community Dance Studio': {
+                        'totals': [10, 0, 0],
+                        'payment_service_fees': [1, 0, 0],
+                        'retained_donations': [9, 0, 0] 
+                    },
+                },
+                'repayments': {
+                    'Power Community Dance Studio': {
+                        'totals': [0, 0, 0],
+                        'revolv_earnings': [0, 0, 0],
+                        'retained_ROI': [0, 0, 0],
+                    },
+                },
+            },
+            'cash_out': {
+                'adjustments': {
+                    'Surprise Expense': [0, 0, 1]
+                },
+                'reinvestments': {
+                    'Power Community Dance Studio': [1, 1, 1],
+                },
+            },
+            'date_info': {
+                'date_list': [
+                    datetime.datetime object,
+                    datetime.datetime object,
+                    datetime.datetime object,
+                ],
+                'start_date': datetime.datetime object,
+                'end_date': datetime.datetime object,
+            },
+            'project_names': ['Power Community Dance Studio'],
+            'cash_balances': {
+                "net_cash_out": [50, 0, 0],
```

Comment:
What is the significance of the three elements of this array? I feel like it may be better structured as a dictionary